### ISSUE TITLE
model not converging the sparse categorical accuracy stays same for the whole epochs. 

### ISSUE THREAD

**Issue by subalbeura** [keras-team/keras-io#1813]
while comparing with the results provided in https://keras.io/examples/timeseries/timeseries_classification_transformer/, the dense layer is different.

```python
dense_2 (Dense)             (None, 128)                  256       ['global_average_pooling1d_2[0
                                                                    ][0]']                        
                                                                                                  
 dropout_25 (Dropout)        (None, 128)                  0         ['dense_2[0][0]']  
```

how to fix it?

**Comment by jentstemmerman** [comment]
Same experience, the model training stops because of the EarlyStopping callback. 
Setting callbacks = None results in 150 EONS of zero progress in accuracy

**Comment by jentstemmerman** [comment]
I think I might have found the solution. The GlobalAveragePooling1D just before the dense layer returns the wrong shape. 
It returns  (None, 1)  with input (None, 500, 1), but the result in the example page is (None, 500)                                 
Changing the parameter data_format into channels_first like below fixes that:     
x = layers.GlobalAveragePooling1D(data_format="channels_first", name="TooSmallOutput")(x)

With that change, the model does converge.

**Comment by subalbeura** [comment]
Without " name="TooSmallOutput" it is also working. 
Regarding TNN, I am also new to the machine learning and am currently trying some examples.

**Comment by jentstemmerman** [comment]
Ah I forgot to remove that.
I added the "name="TooSmallOutput" to debug it.
It just prints out the name in the model summary.

**Comment by subalbeura** [comment]
Thanks for the reply

**Comment by subalbeura** [comment]
For multiclass classification using time series data, the accuracy comes nearly 0.009. What are the factors that can improve accuracy?

**Comment by sonali-kumari1** [comment]
Hi @subalbeura -
Thanks for reporting this issue. Here is a similar issue [#20627](https://github.com/keras-team/keras-io/issues/2395) and a PR [#2006]( https://github.com/keras-team/keras-io/pull/2006) was raised to revert back to `data_format="channels_first` to resolve the accuracy and parameter issues which was introduced by commit https://github.com/keras-team/keras-io/commit/a52a1c5fd913f4861146ea0779acd2c38dc6019f. 

For improving accuracy in multiclass classification using time series data, consider adding more layers to your model, performing a hyperparameter search, optimizing the learning rate schedule, or trying a different optimizer. Additionally, data augmentation and experimenting with different model architectures can help improve performance.
